sdk/go/pulumi/rpc.go

```diff
@@ -61,7 +61,7 @@ import (
 // * Cust4 because it is a child of a custom resource
 // * Comp2 because it is a non-remote component resoruce
 // * Comp3 and Cust5 because Comp3 is a child of a remote component resource
-func addDependency(ctx context.Context, deps urnSet, res, from Resource) error {
+func addDependency(ctx context.Context, deps depSet, res, from Resource) error {
```

Comment:
We need to expand this method to keep track of the resources, not just their URNs. This allows us to iterate over the resources and to await their IDs.